packages/dev/core/src/Engines/thinEngine.ts

```diff
@@ -4379,21 +4379,9 @@ export class ThinEngine {
                 const gl = this._gl;
                 const isPot = img.width === potWidth && img.height === potHeight;
 
-                const internalFormat = format
-                    ? this._getInternalFormat(format, texture._useSRGBBuffer)
-                    : extension === ".jpg" && !texture._useSRGBBuffer
-                    ? gl.RGB
-                    : texture._useSRGBBuffer
-                    ? this._glSRGBExtensionValues.SRGB8_ALPHA8
-                    : gl.RGBA;
-                let texelFormat = format ? this._getInternalFormat(format) : extension === ".jpg" && !texture._useSRGBBuffer ? gl.RGB : gl.RGBA;
-
-                if (texture._useSRGBBuffer && this.webGLVersion === 1) {
-                    texelFormat = internalFormat;
-                }
-
+                const tip = this._getTexImageParametersForCreateTexture(format, extension, texture._useSRGBBuffer);
                 if (isPot) {
-                    gl.texImage2D(gl.TEXTURE_2D, 0, internalFormat, texelFormat, gl.UNSIGNED_BYTE, img as any);
+                    gl.texImage2D(gl.TEXTURE_2D, 0, tip.sizedFormat, tip.internalFormat, tip.type, img as any);

```

Comment:
I think it's a bit misleading to use `internalFormat` for the parameter of `texImage2D` which is named `format` in the docs, whereas the previous one is named `internalFormat`:

![image](https://github.com/BabylonJS/Babylon.js/assets/4152247/06cfeac1-4f5c-4903-980a-de87de328768)

Maybe we should simply use the same names, `internalFormat` and `format`?

Reply:
I've tried changing it such that the naming in TexImageParameters matches the texImage2D function naming, so it's (internalFormat, format, type) rather than (sizedFormat, internalFormat, type).

The downside this way is that elsewhere in the Babylon code "format" means Babylon's format, and "internalFormat" means "what GL calls format", so done this way it's maybe a little confusing from the Babylon end.

I think you're probably right that this is better overall, WDYT?